sql/setup.sql

```diff
@@ -3,42 +3,52 @@ DROP SCHEMA public CASCADE;
 CREATE SCHEMA public;
 
 /* Users table */
-CREATE TABLE users (
-  spotify_userid     VARCHAR(50) PRIMARY KEY,
+CREATE TABLE IF NOT EXISTS users (
+  spotify_userid     VARCHAR(50),
   email              VARCHAR(255),
   authorization_code VARCHAR(255) NOT NULL,
   refresh_token      VARCHAR(255) NOT NULL,
-  access_token       VARCHAR(255)
+  access_token       VARCHAR(255),
+  CONSTRAINT PK_users PRIMARY KEY (spotify_userid),
+  CONSTRAINT UQ_users UNIQUE (email, authorization_code, refresh_token, access_token)
```

Comment:
let's not make this assumption. none of these is necessarily unique (can potentially have multiple emails/account)